ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx

```diff
@@ -0,0 +1,206 @@
+    const timer = sinon.useFakeTimers();
+    const {container} = renderWithContext(<PropertyFilterBuilderRuleGroupRenderer
+      {...defaultProps}
+      group={{id: "id", groupId: "parentId", items: [], operator: PropertyFilterRuleGroupOperator.And}}
+    />);
+
+    expect(container.querySelector(".rule-group[data-isactive=true]")).to.be.null;
+
+    const ruleOperator = container.querySelector(".rule-group-operator");
+    expect(ruleOperator).to.not.be.null;
+    fireEvent.mouseOver(ruleOperator!);
+
+    timer.next();
+
+    expect(container.querySelector(".rule-group[data-isactive=true]")).to.not.be.null;
+    fireEvent.mouseOut(ruleOperator!);
+
+    timer.next();
+
+    expect(container.querySelector(".rule-group[data-isactive=true]")).to.be.null;
+    timer.restore();
+  });
+
+  it("renders child rule", () => {
+    const property: PropertyDescription = {displayLabel: "Prop", name: "prop", typename: "int"};
+    const {getByDisplayValue} = renderWithContext(<PropertyFilterBuilderRuleGroupRenderer
+      {...defaultProps}
+      group={{
+        id: "id",
+        operator: PropertyFilterRuleGroupOperator.And,
+        items: [{
+          id: "childId",
+          groupId: "id",
+          property,
+        }],
+      }}
+    />, {properties: [property]});
+
+    getByDisplayValue("Prop");
+  });
+
+  it("renders child rule group", () => {
+    const {getByText} = renderWithContext(<PropertyFilterBuilderRuleGroupRenderer
+      {...defaultProps}
+      group={{
+        id: "id",
+        operator: PropertyFilterRuleGroupOperator.And,
+        items: [{
+          id: "childId",
+          groupId: "id",
+          operator: PropertyFilterRuleGroupOperator.Or,
+          items: [],
+        }],
+      }}
+    />);
+
+    getByText(TestUtils.i18n.getLocalizedString("Components:filterBuilder.operators.or"));
+  });
+
+  it("dispatches add rule event when button is clicked", () => {
+    const actions = new PropertyFilterBuilderActions(sinon.spy());
+    const {getByTestId} = renderWithContext(<PropertyFilterBuilderRuleGroupRenderer
+      {...defaultProps}
+    />, {actions});
+    const addItemSpy = sinon.stub(actions, "addItem");
+
+    const addRuleButton = getByTestId("rule-group-add-rule");
+    addRuleButton.click();
+
+    expect(addItemSpy).to.be.calledOnceWith(defaultProps.path, "RULE");
+  });
+
+  it("dispatches add rule group event when button is clicked", () => {
+    const actions = new PropertyFilterBuilderActions(sinon.spy());
+    const {getByTestId} = renderWithContext(<PropertyFilterBuilderRuleGroupRenderer
+      {...defaultProps}
+    />, {actions});
+    const addItemSpy = sinon.stub(actions, "addItem");
+
+    const addRuleGroupButton = getByTestId("rule-group-add-rule-group");
+    addRuleGroupButton.click();
+
+    expect(addItemSpy).to.be.calledOnceWith(defaultProps.path, "RULE_GROUP");
+  });
+
+  it("dispatches remove item event when button is clicked", () => {
```

Comment:
Found a bug:

When a rule is deleted, the rule group remains in highlighted state. It is possible to highlight all rule groups at the same time by deleting a rule from each of them.